src/component/reducer/actions/FiltersActions.ts

```diff
@@ -29,8 +29,11 @@ import { resetSelectedTool } from './ToolsActions';
 
 function shiftSpectrumAlongXAxis(draft: Draft<State>, shiftValue) {
   //apply filter into the spectrum
-  if (draft.activeSpectrum?.id) {
-    const index = draft.activeSpectrum?.index;
+
+  const activeSpectrum =
+    draft.view.spectra.activeSpectra[draft.view.spectra.activeTab];
+  if (activeSpectrum?.id) {
+    const index = activeSpectrum?.index;
```

Comment:
Can you implement a helper which does this as it's expected to be used in a lot of places within the reducer.

Also I think this should throw instead of doing nothing if no active spectrum exists. The application should never call this reducer if no active spectrum exists.

Something like a `getActiveSpectrumOrFail(draft)` helper